### PR TITLE
Add MacOS interop and UBSan C to internal CI

### PR DESCRIPTION
**Pull request by matt-kwong**
Eventually we will need to add UBSan C++, but that can wait until its results aren't noisy. 

Linux interop tests still aren't enabled, but I'd also like to start experimenting with MacOS interop tests, since these only test `cloud_to_cloud`.

### CODE REVIEW

File: tools/internal_ci/macos/grpc_interop.cfg [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Config file for the internal CI (in protobuf text format)
+
+# Location of the continuous shell script in repository.
+build_file: "grpc/tools/internal_ci/macos/grpc_interop.sh"
+timeout_mins: 240
+action {
+  define_artifacts {
+    regex: "**/*sponge_log.xml",
+    regex: "reports/**"
```
Comment (jtattermusch) [inline tools/internal_ci/macos/grpc_interop.cfg]: nit:   I think this needs to be github/grpc/reports/** for it to work. I just checked https://github.com/grpc/grpc/blob/master/tools/internal_ci/linux/grpc_build_artifacts.cfg#L38 and that works well.

File: tools/internal_ci/macos/grpc_interop.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+set -ex
+
+# change to grpc repo root
+cd $(dirname $0)/../../..
+
+source tools/internal_ci/helper_scripts/prepare_build_interop_rc
+
+tools/run_tests/run_interop_tests.py -l objc -s all --use_docker -t -j 1
```
Comment (jtattermusch) [inline tools/internal_ci/macos/grpc_interop.sh]: wrong command.  no --use_docker on mac.
At the same time, without --use_docker you can't use  --server option at all (AFAIR).
I  think the most you can do here is --cloud_to_prod. We might be able to do a limited subset of --cloud_to_prod_auth at some point, but not right now.

File: tools/internal_ci/linux/grpc_interop_badserver_java.sh [hunk 3]
```diff
@@ -36,6 +36,7 @@ export LANG=en_US.UTF-8
 cd $(dirname $0)/../../..
 
 source tools/internal_ci/helper_scripts/prepare_build_linux_rc
+source tools/internal_ci/helper_scripts/prepare_build_interop_rc
 
 tools/run_tests/run_interop_tests.py -l java --use_docker --http2_server_interop $@
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_badserver_java.sh]: why are  badserver_java and badserver_python separate? it's probably fine for now but we'll want to stop job proliferation.

Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_badserver_java.sh]: nit: The $@ is not useful here, remove?

Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_badserver_java.sh]: Also elsewhere.

File: tools/internal_ci/linux/grpc_interop_tocloud.sh [hunk 4]
```diff
@@ -36,5 +36,6 @@ export LANG=en_US.UTF-8
 cd $(dirname $0)/../../..
 
 source tools/internal_ci/helper_scripts/prepare_build_linux_rc
+source tools/internal_ci/helper_scripts/prepare_build_interop_rc
 
 tools/run_tests/run_interop_tests.py -l all -s all --use_docker --http2_interop -t -j 12 $@
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_tocloud.sh]: In order to get interop tests building in a reasonable amount of time, you'll probably need to implement the DOCKERHUB_ORGANIZATION trick here:
https://github.com/grpc/grpc/blob/master/tools/run_tests/dockerize/build_interop_image.sh#L83

(and upload the interop base images to dockerhub).

File: tools/internal_ci/macos/grpc_interop.cfg (hunk 1 above)
Comment (jtattermusch) [inline tools/internal_ci/macos/grpc_interop.cfg]: Also fix the report location for the linux interop jobs please.

**Comment by matt-kwong** [comment]
All comments addressed except for 

> wrong command. no --use_docker on mac.
> At the same time, without --use_docker you can't use --server option at all (AFAIR).
> I think the most you can do here is --cloud_to_prod. We might be able to do a limited subset of --cloud_to_prod_auth at some point, but not right now.

This command comes from https://github.com/grpc/grpc/blob/master/tools/jenkins/run_interop_objc.sh written by @makdharma.

I've been working on getting this to work on one of the Jenkins Mac workers and I am currently stuck on a non-Docker related issue, but Docker on the Mac seems to work fine with Docker Machine. AFAIK, it should be possible to run Obj-C locally and use Docker for the other languages acting as the server. I'm adding this internally to see if our internal environment might be easier to get this test running on.

**Comment by jtattermusch** [comment]
Not sure I'm happy about running Docker on the mac workers (seems like a fairly heavyweight solution). It might make more sense to actually get run_interop_tests.py working with non-docker servers (but maybe not).

**Comment by makdharma** [comment]
For per language interop testing between ObjC and all languages I could not come up w

**Comment by makdharma** [comment]
Sorry closed by mistake. disregard previous comment.

**Comment by matt-kwong** [comment]
I'll try to get interop MacOS working internally, and if it's not working, I'll work with Makarand to see if it's possible to run this test without Docker.

All tests were passing before resolving merge conflict.

**Pull request merged**